Formula/fish.rb

```diff
@@ -22,9 +22,16 @@ class Fish < Formula
 
   def install
     system "autoconf" if build.head? || build.devel?
+    args = %W[
+      --prefix=#{prefix}
+      --with-extra-functionsdir=#{HOMEBREW_PREFIX}/share/fish/vendor_functions.d
```

Comment:
> #{pkgshare} will not work. The patch is about having version independent dirs outside of the Cellar so other Formulae can install completions and functions into that dir and fish can use those.
By default fish looking only into own Cellar share dirs for scripts. My PR is trying to make fish to look into #{HOMEBREW_PREFIX}/share/fish for completions and functions.

Yeah. By definition it wouldn't be sensible to use `pkgshare` here. It does make sense to use a real directory in `HOMEBREW_PREFIX`. We're doing something similar enough [already](https://github.com/Homebrew/brew/blob/c6022104f9df7fa1fc7bae596a770cdc073c8552/Library/Homebrew/keg.rb#L315-L324).

Does `fish` actually put anything in that directory at compile time, or is it purely used to teach `fish` to look for functions there as well as the default location?